lib/lintPostcssResult.mjs

```diff
@@ -137,5 +138,15 @@ function isFixCompatible({ stylelint: { disabledRanges } }, name) {
  * @returns {void}
  */
 function warnDeprecatedRule(result, ruleName) {
```

Comment:
off-topic

I am not sure, but it seems like this function may produces duplicates warnings for plugins.
e.g. https://github.com/stylelint-scss/stylelint-scss/blob/a689b55e41a4c582377f2b291236757e4594b429/src/rules/at-import-partial-extension/index.js#L51-L59

If that's the case, it might be wiser to modify it, in another PR, so that it only warns for core rules.

Reply:
This point is still important. The current Stylelint collects deprecation warnings in a PostCSS result and shows them through the `string` formatter:

https://github.com/stylelint/stylelint/blob/30e1027b93520193b73a692394db124132144ba5/lib/createPartialStylelintResult.mjs#L25-L33

https://github.com/stylelint/stylelint/blob/30e1027b93520193b73a692394db124132144ba5/lib/formatters/stringFormatter.mjs#L45-L47

So, I think we have to fix the formatter code in this PR to work `--no-deprecation` and leave `result.warn()` as-is here to keep backward compatibility of a lint result.

Reply:
I don't understand your point.
I am saying we might have 2 warnings for each deprecation.
Am I correct in my assumption?
If so, I am proposing to fix it in another PR.
This PR is about adding `006` for rule deprecation and suppressing `warn` when `noDeprecation` is `true`.

Reply:
Are you proposing to add another `if (process.noDeprecation)`?

Reply:
I moved the `if (!process.noDeprecation)` in `createPartialStylelintResult.mjs`.